Tasks/Common/PowershellHelpers/Tests/Invoke-ActionWithRetries.ShouldHonorMaxRetriesParameter.ps1

```diff
@@ -0,0 +1,20 @@
+[CmdletBinding()]
+param()
+
+# Arrange.
+. $PSScriptRoot\..\..\..\..\Tests\lib\Initialize-Test.ps1
+$module = Microsoft.PowerShell.Core\Import-Module $PSScriptRoot\.. -PassThru
+$global:retriesAttempted = 0
+$action = {
+    $global:retriesAttempted++
+    throw [System.IO.FileNotFoundException] "File not found error!"
+}
+
+Register-Mock Set-UserAgent
+Unregister-Mock Start-Sleep
+Register-Mock Start-Sleep {}
+# Act/Assert.
+Assert-Throws {
+    & $module Invoke-ActionWithRetries -Action $action -MaxRetries 5
+} -MessagePattern "File not found error!"
+Assert-AreEqual 5 $global:retriesAttempted "Number of retries not correct"
```

Comment:
That would be the number of tries, retrying by definition requires having already tried again. Please rename to MaxTries.